### DISCUSSION TITLE
2 ebusd V5 adapters into home assistant?

### DISCUSSION THREAD

**Discussion by Infernoken76** [john30/ebusd#1223] [General]
Hi,
I have situation where I use a gasboiler Ecotec Vaillant VC126 and a heatpump aroTHERM Plus 75/6...
(with a VRC720 VR70,...)

Both of them work together with an ebus bridge VR32 and right now I have a ebus V5 dongle working with the heatpump, the VRC720 & the VR70...
But it cannot see the gasboiler as it's behind the VR32...

So I bought a second ebus V5 dongle to monitor the gasboiler...

But how do I get this data into home assistant?

I already have the ebusd add-on running for my first dongle (https://github.com/LukasGrebe/ha-addons)...
Do you know how to integrate 2 dongles into home assistant?

**Comment by JonesGW-MGD** [comment]
@Infernoken76 - I don't know the answer to your question, but you can usually access the boiler behind the VR32 - is there a reason you don't want to do this?

I seem to have helped a couple of people do this recently (not sure how - I don't have a VR32!). 

It basically involves - 

1. Cloning config files locally
2. Ensuring you use --latency=100000 at startup
3. Establishing which address your VR32 presents itself to the network on (as slave) - this generally seems to be 18, 38 or 78 but YMMV
4. Finding an appropriate bai.inc file for your boiler - you can start with bai308523.inc (unless manually reading through 08.bai.csv suggests a better fit)
5. Rename (or symlink) this so you have a [Address].V32.csv file
6. Ebusd should find and use this on auto scan 

Have a look here https://github.com/john30/ebusd/issues/1221 or here: https://github.com/john30/ebusd/discussions/1197#discussioncomment-8543829

**Reply by Infernoken76** [reply]
Thanks, tried this, but the ebusd add-on in home assistant doesn't allow for such high latency?

This is the error I get:

Failed to save add-on configuration, value must be at most 10000.0. Got {'scanconfig': True, 'loglevel_all': 'notice', 'mqtttopic': 'ebusd', 'mqttint': '/config/ebusd-configuration/mqtt-hassio.cfg', 'mqttjson': True, 'configpath': '/config/ebusd-configuration/', 'accesslevel': '*', 'readonly': False, 'network_device': 'enh:192.168.1.61:9999', 'latency': 100000, 'pollinterval': 4}

**Comment by JonesGW-MGD** [comment]
Well that made me curious!

I also run the ebusd add-on.

So I shut my config down, altered it by adding latency and started it again and it seems fine...

```
6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
> ebusd --foreground --httpport=8889 --mqtthost=core-mosquitto --mqttport=1883 --mqttuser=addons --mqttpass=xxx --scanconfig --mqttjson --configpath=/config/ebusd-configuration/latest/en/ --mqttint=/etc/ebusd/mqtt-hassio.cfg --mqtttopic=ebusd --device=enh:/dev/ttyACM0 --log=all:notice --log=main:notice --log=other:notice --mqttvar=filter-name= --mqttvar=filter-direction=r|u|^w --latency=100000
2024-03-11 17:59:34.005 [main notice] ebusd 23.2.p20230716 started with auto scan on device: /dev/ttyACM0, enhanced
```

This is my (temporary!) line in the Custom Command Line Options part of the add-on configuration (the last command being relevant to you):

```
--mqttvar=filter-name= --mqttvar=filter-direction=r|u|^w --latency=100000
```
Is it possible you just keyed it incorrectly?

**Reply by Infernoken76** [reply]
Thanks, it's now working!
I see my gasboiler with the adaptation of bai308523.inc to 18.V32.csv

I had the latency defined in the settings of ebusd config and not via the Custom Command Line Options...  Now it accepts the latency of 100000

**Reply by JonesGW-MGD** [reply]
Great!  All the best and good luck!

**Comment by lennycb** [comment]
@JonesGW-MGD 

Because it fits well, I would warm up the topic again.

Since I have a Vaillant device behind a VR32, I got a second adapter V5.
I have a problem with the multiple instances of ebusd.

Now I read this and tried it straight away.
Unfortunately it doesn't work that way for me.

Settings:
/etc/default/ebusd
EBUSD_OPTS="--configpath=/etc/ebusd/dateien/de --scanconfig=full --configlang=de --device=ens:192.168.0.47:9999 --enablehex --latency=100000 --receivetimeout=100000 --httpport=8889 --htmlpath=/var/ebusd/html"

What have I done:
"08.bai.csv" copied and renamed to "18.V32.csv"	

service ebusd stop
service ebusd start

Result:

root@eBUSd:~# ebusctl info
version: ebusd 23.2.23.2
update check: version 23.3 available, device firmware 1[430a] available, vaillant/08.hmu.csv: newer version available, vaillant/15.700.csv: newer version available
device: 192.168.0.47:9999, enhanced, firmware 1.1[3a0f].1[3a0f]
signal: acquired
symbol rate: 115
max symbol rate: 168
min arbitration micros: 0
max arbitration micros: 32
min symbol latency: 10
max symbol latency: 43
scan: finished
reconnects: 0
masters: 5
messages: 633
conditional: 0
poll: 1
update: 10
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0607;HW=5103", loaded "vaillant/08.hmu00.csv"
address 10: master #2
address 13: master #12
address 15: slave #2, scanned "MF=Vaillant;ID=70000;SW=0614;HW=6903", loaded "vaillant/15.700.csv"
address 18: slave #12, scanned "MF=Vaillant;ID=V32;SW=0106;HW=6004"
address 26: slave, scanned "MF=Vaillant;ID=VR_71;SW=0203;HW=5904", loaded "vaillant/26.vr_71.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd, scanning
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0202;HW=0103"
address ec: slave, scanned "MF=Vaillant;ID=SOL00;SW=0614;HW=6903", loaded "vaillant/ec.sol.sc.csv"

what am I doing wrong?

**Reply by JonesGW-MGD** [reply]
@lennycb - I suspect you're very close!  I think you may have been tripped up by the fact that 08.bai.csv is essentially a manifest file - so differs from say 08.hmu00.csv which has actual message definitions in it.

You'll need to explore the bai.xxxxxx.inc files to find an appropriate one if indeed your device behind the V32 is a gas boiler.   To start with, why don't you try bai.308523.inc which is essentially the default that would ordinarily get loaded if the auto configuration process parses through 08.bai.csv and can't find a more appropriate file (by Product ID or HW id) at start up

So rename that one to 18.v32.csv and see if that helps.

**Comment by lennycb** [comment]
From the looks of it, it's working now. Thank you!

address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0607;HW=5103", loaded "vaillant/08.hmu00.csv"
address 10: master #2
address 13: master #12
address 15: slave #2, scanned "MF=Vaillant;ID=70000;SW=0614;HW=6903", loaded "vaillant/15.700.csv"
address 18: slave #12, scanned "MF=Vaillant;ID=V32;SW=0106;HW=6004", loaded "vaillant/bai.308523.inc", "vaillant/18.v32.csv"
address 26: slave, scanned "MF=Vaillant;ID=VR_71;SW=0203;HW=5904", loaded "vaillant/26.vr_71.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd, scanning
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0202;HW=0103"
address ec: slave, scanned "MF=Vaillant;ID=SOL00;SW=0614;HW=6903", loaded "vaillant/ec.sol.sc.csv"

**Reply by JonesGW-MGD** [reply]
Great!  @lennycb  - if you’re feeling lucky, there isn’t an ‘official’ repo file for the vwzio device on address 76 (is that something like the electrically powered hydraulic connection unit VWZ MEH 61?) but there are ‘experimental’ files in forked repositories - you could try here for example:

https://github.com/rmalbrecht/ebusd-configuration/blob/master/ebusd-2.1.x/en/vaillant/76.vwzio.csv

Don’t know whether you would get any more useful info that isn’t available elsewhere on your ebus though.
